tests/testOperatorInterface.py

```diff
@@ -764,6 +764,31 @@ def propagateDirty(self, *a, **kw):
         op_b.setupOutputs.assert_called_once()
 
 
+class TestCompatibilityChecks:
+    class OpA(graph.Operator):
+        Output = graph.OutputSlot()
+
+        def setupOutputs(self):
+            self.Output.meta.shape = (3, 3)
+            self.Output.meta.dtype = int
+
+        def propagateDirty(self, *a, **kw):
+            pass
+
+        def execute(self, *args, **kwargs):
+            return numpy.ones((2, 2), dtype=int)
+
+    def test_arraylike_raises_if_shapes_are_mismatched(self, graph):
+        op = self.OpA(graph=graph)
+
+        with pytest.raises(Exception):
```

Comment:
ugh, this is all done with asserts, at least in `ArrayLike`.  Should probably check for `AssertionError`, slightly more specific. But I guess it's all wrong that it is done with asserts in the first place.